{"relationshipsSections":[{"identifiers":["doc:\/\/SwiftKyber\/s5ErrorP","doc:\/\/SwiftKyber\/s8SendableP"],"title":"Conforms To","kind":"relationships","type":"conformsTo"}],"primaryContentSections":[{"kind":"declarations","declarations":[{"platforms":["macOS"],"languages":["swift"],"tokens":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"text":"KyberException","kind":"identifier"}]}]}],"identifier":{"interfaceLanguage":"swift","url":"doc:\/\/SwiftKyber\/documentation\/SwiftKyber\/KyberException"},"hierarchy":{"paths":[["doc:\/\/SwiftKyber\/documentation\/SwiftKyber"]]},"kind":"symbol","variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/swiftkyber\/kyberexception"]}],"abstract":[{"type":"text","text":"The Kyber exceptions"}],"metadata":{"navigatorTitle":[{"text":"KyberException","kind":"identifier"}],"title":"KyberException","roleHeading":"Enumeration","modules":[{"name":"SwiftKyber"}],"symbolKind":"enum","externalID":"s:10SwiftKyber0B9ExceptionO","role":"symbol","fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"text":"KyberException","kind":"identifier"}]},"schemaVersion":{"major":0,"patch":0,"minor":3},"topicSections":[{"title":"Enumeration Cases","identifiers":["doc:\/\/SwiftKyber\/documentation\/SwiftKyber\/KyberException\/cipherTextSize(value:)","doc:\/\/SwiftKyber\/documentation\/SwiftKyber\/KyberException\/decapsulationKeyInconsistent","doc:\/\/SwiftKyber\/documentation\/SwiftKyber\/KyberException\/decapsulationKeySize(value:)","doc:\/\/SwiftKyber\/documentation\/SwiftKyber\/KyberException\/encapsulationKeyInconsistent","doc:\/\/SwiftKyber\/documentation\/SwiftKyber\/KyberException\/encapsulationKeySize(value:)"]},{"generated":true,"identifiers":["doc:\/\/SwiftKyber\/documentation\/SwiftKyber\/KyberException\/Error-Implementations"],"title":"Default Implementations"}],"sections":[],"references":{"doc://SwiftKyber/documentation/SwiftKyber/KyberException/Error-Implementations":{"role":"collectionGroup","title":"Error Implementations","url":"\/documentation\/swiftkyber\/kyberexception\/error-implementations","identifier":"doc:\/\/SwiftKyber\/documentation\/SwiftKyber\/KyberException\/Error-Implementations","type":"topic","kind":"article","abstract":[]},"doc://SwiftKyber/documentation/SwiftKyber/KyberException/encapsulationKeySize(value:)":{"title":"KyberException.encapsulationKeySize(value:)","url":"\/documentation\/swiftkyber\/kyberexception\/encapsulationkeysize(value:)","type":"topic","role":"symbol","abstract":[{"text":"Wrong encapsulation key size","type":"text"}],"fragments":[{"text":"case","kind":"keyword"},{"text":" ","kind":"text"},{"text":"encapsulationKeySize","kind":"identifier"},{"text":"(","kind":"text"},{"text":"value","kind":"externalParam"},{"text":": ","kind":"text"},{"text":"Int","kind":"typeIdentifier","preciseIdentifier":"s:Si"},{"kind":"text","text":")"}],"identifier":"doc:\/\/SwiftKyber\/documentation\/SwiftKyber\/KyberException\/encapsulationKeySize(value:)","kind":"symbol"},"doc://SwiftKyber/s5ErrorP":{"title":"Swift.Error","type":"unresolvable","identifier":"doc:\/\/SwiftKyber\/s5ErrorP"},"doc://SwiftKyber/documentation/SwiftKyber/KyberException":{"abstract":[{"text":"The Kyber exceptions","type":"text"}],"role":"symbol","navigatorTitle":[{"text":"KyberException","kind":"identifier"}],"identifier":"doc:\/\/SwiftKyber\/documentation\/SwiftKyber\/KyberException","kind":"symbol","fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"KyberException"}],"type":"topic","url":"\/documentation\/swiftkyber\/kyberexception","title":"KyberException"},"doc://SwiftKyber/documentation/SwiftKyber/KyberException/decapsulationKeySize(value:)":{"title":"KyberException.decapsulationKeySize(value:)","abstract":[{"text":"Wrong decapsulation key size","type":"text"}],"identifier":"doc:\/\/SwiftKyber\/documentation\/SwiftKyber\/KyberException\/decapsulationKeySize(value:)","url":"\/documentation\/swiftkyber\/kyberexception\/decapsulationkeysize(value:)","fragments":[{"text":"case","kind":"keyword"},{"text":" ","kind":"text"},{"text":"decapsulationKeySize","kind":"identifier"},{"text":"(","kind":"text"},{"text":"value","kind":"externalParam"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:Si","text":"Int","kind":"typeIdentifier"},{"text":")","kind":"text"}],"kind":"symbol","type":"topic","role":"symbol"},"doc://SwiftKyber/documentation/SwiftKyber":{"type":"topic","abstract":[{"text":"Module-Lattice-based Key-Encapsulation Mechanism Standard","type":"text"}],"role":"collection","kind":"symbol","url":"\/documentation\/swiftkyber","title":"SwiftKyber","identifier":"doc:\/\/SwiftKyber\/documentation\/SwiftKyber"},"doc://SwiftKyber/documentation/SwiftKyber/KyberException/encapsulationKeyInconsistent":{"title":"KyberException.encapsulationKeyInconsistent","url":"\/documentation\/swiftkyber\/kyberexception\/encapsulationkeyinconsistent","role":"symbol","type":"topic","kind":"symbol","identifier":"doc:\/\/SwiftKyber\/documentation\/SwiftKyber\/KyberException\/encapsulationKeyInconsistent","abstract":[{"text":"Inconsistent encapsulation key data","type":"text"}],"fragments":[{"kind":"keyword","text":"case"},{"text":" ","kind":"text"},{"kind":"identifier","text":"encapsulationKeyInconsistent"}]},"doc://SwiftKyber/documentation/SwiftKyber/KyberException/cipherTextSize(value:)":{"fragments":[{"text":"case","kind":"keyword"},{"text":" ","kind":"text"},{"text":"cipherTextSize","kind":"identifier"},{"kind":"text","text":"("},{"text":"value","kind":"externalParam"},{"kind":"text","text":": "},{"preciseIdentifier":"s:Si","text":"Int","kind":"typeIdentifier"},{"text":")","kind":"text"}],"kind":"symbol","title":"KyberException.cipherTextSize(value:)","url":"\/documentation\/swiftkyber\/kyberexception\/ciphertextsize(value:)","abstract":[{"text":"Wrong cipher text size","type":"text"}],"role":"symbol","type":"topic","identifier":"doc:\/\/SwiftKyber\/documentation\/SwiftKyber\/KyberException\/cipherTextSize(value:)"},"doc://SwiftKyber/documentation/SwiftKyber/KyberException/decapsulationKeyInconsistent":{"kind":"symbol","title":"KyberException.decapsulationKeyInconsistent","identifier":"doc:\/\/SwiftKyber\/documentation\/SwiftKyber\/KyberException\/decapsulationKeyInconsistent","abstract":[{"type":"text","text":"Inconsistent decapsulation key data"}],"url":"\/documentation\/swiftkyber\/kyberexception\/decapsulationkeyinconsistent","role":"symbol","fragments":[{"text":"case","kind":"keyword"},{"text":" ","kind":"text"},{"text":"decapsulationKeyInconsistent","kind":"identifier"}],"type":"topic"},"doc://SwiftKyber/s8SendableP":{"identifier":"doc:\/\/SwiftKyber\/s8SendableP","title":"Swift.Sendable","type":"unresolvable"}}}